src/components/application_manager/test/state_controller/state_controller_test.cc

```diff
@@ -2490,6 +2564,422 @@ TEST_F(StateControllerImplTest,
   state_ctrl_->SetRegularState(media_app_, check_state, false);
 }
 
+TEST_F(StateControllerImplTest,
+       SetStateForNaviApp_BCOnPhoneCall_SetPostponedState) {
+  using namespace hmi_apis;
+  using namespace smart_objects;
+  using namespace am::event_engine;
+
+  // Precondition
+  Event event(FunctionID::BasicCommunication_OnEventChanged);
+  SmartObject message;
+  message[am::strings::msg_params][am::hmi_notification::is_active] = true;
+  message[am::strings::msg_params][am::hmi_notification::event_name] =
+      hmi_apis::Common_EventTypes::PHONE_CALL;
+
+  event.set_smart_object(message);
+  state_ctrl_->on_event(event);
+
+  am::HmiStatePtr hmi_state = FullAudibleState();
+
+  EXPECT_CALL(*navi_app_ptr_, is_resuming()).WillRepeatedly(Return(true));
+
+  EXPECT_CALL(*navi_app_ptr_, is_navi()).WillRepeatedly(Return(true));
+
+  EXPECT_CALL(app_manager_mock_, IsAppTypeExistsInFullOrLimited(_))
+      .WillRepeatedly(Return(false));
+
+  EXPECT_CALL(*navi_app_ptr_, CurrentHmiState())
+      .WillRepeatedly(Return(hmi_state));
+
+  EXPECT_CALL(app_manager_mock_, active_application())
+      .WillRepeatedly(Return(am::ApplicationSharedPtr()));
+
+  state_ctrl_->SetRegularState(navi_app_, hmi_state, false);
+}
+
+TEST_F(StateControllerImplTest,
+       SetStateForNaviApp_BCOnPhoneCall_NotPostponedState) {
+  using namespace hmi_apis;
+  using namespace smart_objects;
+  using namespace am::event_engine;
+
+  // Precondition
+  Event event(FunctionID::BasicCommunication_OnEventChanged);
+  SmartObject message;
+  message[am::strings::msg_params][am::hmi_notification::is_active] = false;
+  message[am::strings::msg_params][am::hmi_notification::event_name] =
+      hmi_apis::Common_EventTypes::PHONE_CALL;
+
+  event.set_smart_object(message);
+  state_ctrl_->on_event(event);
+
+  am::HmiStatePtr hmi_state = FullAudibleState();
+
+  EXPECT_CALL(*navi_app_ptr_, is_resuming()).WillRepeatedly(Return(true));
+
+  EXPECT_CALL(*navi_app_ptr_, is_navi()).WillRepeatedly(Return(true));
+
+  EXPECT_CALL(app_manager_mock_, IsAppTypeExistsInFullOrLimited(_))
+      .WillRepeatedly(Return(false));
+
+  EXPECT_CALL(*navi_app_ptr_, CurrentHmiState())
+      .WillRepeatedly(Return(hmi_state));
+
+  EXPECT_CALL(app_manager_mock_, active_application())
+      .WillRepeatedly(Return(am::ApplicationSharedPtr()));
+
+  EXPECT_CALL(*navi_app_ptr_, SetPostponedState(_)).Times(0);
+
+  state_ctrl_->SetRegularState(navi_app_, hmi_state, false);
+}
+
+TEST_F(StateControllerImplTest,
+       SetStateForNaviApp_BCOnPhoneCall_SetPostponedStateWithActivation) {
+  using namespace hmi_apis;
+  using namespace smart_objects;
+  using namespace am::event_engine;
+
+  const uint32_t app_id = navi_app_->app_id();
```

Comment:
@pvvasilev, incorrect `const` naming. Please, check naming of all `const` variables.